internal/cmd/launcher.go

```diff
@@ -277,37 +276,37 @@ func extractToken(gs *state.GlobalState) (string, error) {
 // Presently, only the k6 input script or archive (if any) is passed to k6deps for scanning.
 // TODO: if k6 receives the input from stdin, it is not used for scanning because we don't know
 // if it is a script or an archive
-func analyze(gs *state.GlobalState, args []string) (k6deps.Dependencies, error) {
+func analyze(gs *state.GlobalState, _ *cobra.Command, args []string) (k6deps.Dependencies, error) {
 	dopts := &k6deps.Options{
 		LookupEnv: func(key string) (string, bool) { v, ok := gs.Env[key]; return v, ok },
 		Manifest:  k6deps.Source{Ignore: true},
 	}
 
-	if len(args) == 0 {
-		gs.Logger.
-			Debug("The command did not receive an input script.")
-		return nil, errScriptNotFound
+	if len(args) < 1 {
+		return nil, fmt.Errorf("k6 needs at least one argument to load the test")
 	}
 
-	scriptname := args[0]
-	if scriptname == "-" {
-		gs.Logger.
-			Debug("Test script provided by Stdin is not yet supported from Binary provisioning feature.")
-		return nil, errUnsupportedFeature
+	sourceRootPath := args[0]
+	gs.Logger.Debugf("Resolving and reading test '%s'...", sourceRootPath)
+	src, _, pwd, err := readSource(gs, sourceRootPath)
```

Comment:
I don't like the idea to have this function called twice. Can we move this function as early as possible to be good for the entire k6 program? Independently if Binary Provisioning is enabled or not. In this way, the part after can assume it is always already loaded.

I don't know if it is really possible, but I want to make sure that we thought about it and we have reasons if we have to call it twice.

Reply:
I will open another PR trying to address this particular issue as it probably will touch many places in the code inderectly.

Reply:
@pablochacin AFAIK, this isn't going to be addressed, can you please open an issue so we can track it and discuss in async on the various options, please?